magma/syntax/sequential.py

```diff
@@ -171,21 +174,19 @@ def __call__(self, I):
 
 
 circuit_definition_template = """
-from magma import In, Out, Bit
-
-def make_{circuit_name}(combinational):
-    class {circuit_name}(m.Circuit):
-        IO = {io_list}
-
-        @classmethod
-        def definition(io):
-            {register_instances}
-            @combinational
-            def {circuit_name}_comb({circuit_combinational_args}) -> ({circuit_combinational_output_type}):
-                {circuit_combinational_body}
-            comb_out = {circuit_name}_comb({circuit_combinational_call_args})
-            {comb_out_wiring}
-    return {circuit_name}
+class {circuit_name}(m.Circuit):
+    IO = {io_list}
+
+    @classmethod
+    def definition(io):
+        {register_instances}
+        @ast_tools.passes.end_rewrite()
+        @m.circuit.combinational()
+        @ast_tools.passes.begin_rewrite()
+        def {circuit_name}_comb({circuit_combinational_args}) -> ({circuit_combinational_output_type}):
```

Comment:
Could we build the AST and directly call `m.circuit.combinational()`?  This would avoid needing to inject ast_tools into env.